src/Button/Button.tsx

```diff
@@ -72,46 +72,9 @@ export const Button: FC<ButtonProps> = forwardRef<
       ...props
     },
     ref,
-  ) => (
-    <div>
-      {primary || secondary || tertiary ? (
-        <Container
-          id={id}
-          className={className}
-          disabled={disabled || loading}
-          onClick={(e) => {
-            handleClick && handleClick(e)
-          }}
-          isLoading={loading}
-          primary={primary}
-          secondary={secondary}
-          tertiary={tertiary}
-          icon={icon}
-          forcedWidth={forcedWidth}
-          {...(form ? { form } : {})}
-          type={type}
-          {...props}
-          ref={ref}
-        >
-          {loading ? (
-            <Loader
-              color={primary ? 'white' : `${theme.colors.secondary}`}
-              height="16"
-            />
-          ) : (
-            <>
-              {icon && (
-                <IconContainer
-                  render={icon}
-                  size={24}
-                  color={primary ? 'white' : 'secondary'}
-                />
-              )}
-              <ChildrenContainer>{children}</ChildrenContainer>
-            </>
-          )}
-        </Container>
-      ) : (
+  ) => {
+    if (!primary && !secondary && !tertiary) {
+      return (
         <LegacyButton
           id={id}
           className={className}
```

Comment:
Moved the legacy button to an early return to remove a bit of noise from the return element

Reply:
I'm hoping at some point to get rid of that legacy button - I think we are in a good place but still not 100% sure